apps/aecore/lib/aecore/peers/worker.ex

```diff
@@ -92,7 +99,31 @@ defmodule Aecore.Peers.Worker do
     {:ok, initial_peers}
   end
 
-  def handle_call({:add_peer, uri}, _from, state) do
+  def handle_call(:is_chain_synced, _from, %{peers: peers} = state) do
+    local_latest_block_height = Chain.latest_block().header.height
+    peer_uris = peers
+      |> Map.values()
+      |> Enum.map(fn(%{uri: uri}) -> uri end)
+    peer_latest_block_heights =
+      Enum.map(peer_uris, fn(uri) ->
+          case Client.get_info(uri) do
```

Comment:
We actually store in state latest_block_info for every peer. Please use that instead so we won't make any requests when handling this call.

Reply:
yeah, good find!

Reply:
We only store latest_block_hash, in the state, if we don't get the info here we would have to do check_peers before mining which would do these requests anyway, I think it's fine like it is now but i'm open to suggestions

Reply:
@cytadela8 should be fine